File: Fresh/Fresh.Model/Requests/PurchaseInsertRequest.cs
namespace Fresh.Model.Requests
{
    public class PurchaseInsertRequest
    {
        public decimal Quantity { get; set; }
        public int ClientId { get; set; }
        public int ProductId { get; set; }
        public PaymentType PaymentType { get; set; }
        public int? NumberOfInstallments { get; set; }
    }
}

File: Fresh/UI/fresh_app/src/app/model/requests/PurchaseInsertRequest.ts
import { PaymentType } from "../PaymentType";

export class PurchaseInsertRequest {
    quantity: number;
    clientId: number;
    productId: number;
    paymentType: PaymentType;
    numberOfInstallments?: number;

    constructor(quantity: number,
        clientId: number,
        productId: number,
        paymentType: PaymentType,
        numberOfInstallments: number | undefined
    ) {
        this.quantity = quantity;
        this.clientId = clientId;
        this.productId = productId;
        this.paymentType = paymentType;
        this.numberOfInstallments = numberOfInstallments;
    }

}
